python-package/lightgbm/dask.py

```diff
@@ -593,14 +755,23 @@ def fit(
         X: _DaskMatrixLike,
         y: _DaskCollection,
         sample_weight: Optional[_DaskCollection] = None,
+        init_score: Optional[_DaskCollection] = None,
+        eval_set: Optional[List[Tuple[_DaskCollection, _DaskCollection]]] = None,
+        eval_sample_weight: Optional[List[_DaskCollection]] = None,
```

Comment:
could you add `eval_class_weight` right here, and then raise an exception if it's non-None and says that it isn't yet supported? I don't think it should be in this PR and I can write up a feature request for it.

But I'd like these methods to have the same argument positions as `lgb.sklearn.LGBMModel.fit()`:

https://github.com/microsoft/LightGBM/blob/fd094d2719a04b984b376a665b6c3e5e1aff2967/python-package/lightgbm/sklearn.py#L545

and can you please do this in `DaskLGBMRegressor.fit()` and `DaskLGBMRanker.fit()` as well?

Reply:
Yeah for sure. Should I add `eval_metric` and `early_stopping_rounds` too, or just assume they're part of `kwargs`?

Reply:
oh right! Yes please, adding them here would be good and I think it's in scope for this PR

Reply:
Addressed in 18cd42c